bin/main.rb

```diff
@@ -1,2 +1,110 @@
-#!/usr/bin/env ruby
-puts 'Hello world!'
+# rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+# !/usr/bin/env ruby
+
+require_relative '../lib/classes.rb'
+
+puts 'Welcome to our Tic-Tac-Toe Game!'
+
+puts 'Player 1, please type your beautiful name?'
+player_one_name = gets.chomp
+while player_one_name == ''
+  puts 'Hey, please, tell us your name :)'
+  player_one_name = gets.chomp
+end
+player_one = Player.new(player_one_name, 'X')
+puts "Nice name #{player_one.name}!"
+puts 'Players, you are going to be represented by the "X" characters on the board, right? (Press enter to continue)'
+gets.chomp
+
+puts 'Player 2, how could we call this beautiful face?'
+player_two_name = gets.chomp
+while player_two_name == ''
+  puts 'Hey, please, tell us your name :)'
+  player_two_name = gets.chomp
+end
+player_two = Player.new(player_two_name, 'O')
+
+puts "Huuummmm #{player_two.name}, I was pretty aware that your name would be also beautiful!"
+puts "As #{player_one.name} is already using the 'X' character, you are going to be the 'O' character, nice? (Press Enter)"
+gets.chomp
+
+puts "#{player_one.name} and #{player_two.name}, please promise me you won't forget your characters? (Press Enter)"
+gets.chomp
+
+puts "Ok #{player_one.name} and #{player_two.name}, lets to the nuts and bolts!"
+puts 'Here is the board of your game, with the respective number for each position, ok?'
+plays = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+class Gameboard
+  def print_board(plays)
+    puts " #{plays[0]} | #{plays[1]} | #{plays[2]}"
+    puts " #{plays[3]} | #{plays[4]} | #{plays[5]}"
+    puts " #{plays[6]} | #{plays[7]} | #{plays[8]}"
+  end
+end
+
+gameboard_instance = Gameboard.new
+puts gameboard_instance.print_board(plays)
+
+class Gameplay
+  def catch_play(plays)
+    play = play.to_i
+    while plays[play - 1] == 'X' || plays[play - 1] == 'O' || !(play.to_i.positive? && play.to_i <= 9)
+      play = gets.chomp.to_i
+      if plays[play - 1] == 'X' || plays[play - 1] == 'O' || !(play.to_i.positive? && play.to_i <= 9)
+        puts 'Ops, it seems you did not type a number between 1-9 OR someone already choosen this slot before :/'
+      end
+    end
+    play
+  end
+end
```

Comment:
- [ ] All the classes and logic need to be inside the lib folder. For printing the board, you could try just return a string and print the function in the main.rb.